spec/ddtrace/contrib/http/miniapp_spec.rb

```diff
@@ -5,24 +5,26 @@
 require 'time'
 
 RSpec.describe 'net/http miniapp tests' do
-  before { WebMock.enable! }
-
-  after do
-    WebMock.reset!
-    WebMock.disable!
-  end
-
   let(:host) { '127.0.0.1' }
   let(:port) { 1234 }
   let(:uri) { "http://#{host}:#{port}" }
 
-  let(:client) { Net::HTTP.new(host, port) }
+  context 'when performing a trace around HTTP calls' do
+    before { WebMock.enable! }
 
-  before do
-    Datadog.configure { |c| c.use :http }
-  end
+    after do
+      WebMock.reset!
+      WebMock.disable!
+      Datadog::Contrib::HTTP::Patcher.remove_instance_variable(:@done_once) if Datadog::Contrib::HTTP::Patcher.patched?
```

Comment:
`WebMock.disable!` restores the original version of `Net::HTTP` that does not contain the DD patch. Originally we started `WebMock.enable!` after the patch was loaded, so when restoring, it would just restore to the DD patched Net::HTTP.

Because the connection error tests need to remove the Webmock completely, it is reloading the "pure" `Net::HTTP`. Somehow it interferes with all this patching.

There may be nicer/cleaner way to do all this, this is what I ended up with that works :)

Reply:
It would require to properly understand the order of patching during running those specs. If we can guarantee that we always first patch Datadog before any `WebMock.enable!` for the whole `rspec` runtime, then probably we would not need this to force repatching.